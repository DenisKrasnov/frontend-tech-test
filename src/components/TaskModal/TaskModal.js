import React from "react";
import { Modal } from "react-overlays";
import { bool, func, string, shape } from "prop-types";

import Icon from "../Icon";
import TaskForm from "../TaskForm";
import styles from "./TaskModal.scss";

const TaskModal = props => {
  const { isOpen, onClose, task, editTask } = props;

  return (
    <Modal
      backdropClassName={styles.backdrop}
      className={styles.TaskModal}
      onHide={onClose}
      show={isOpen}
    >
      <div className={styles.modal}>
        <div className={styles.header}>
          <h2 className={styles.title}>Edit task</h2>
          <span className={styles.closeButton}>
            <Icon glyph="x" onClick={onClose} />
          </span>
        </div>
        <div className={styles.body}>
          <TaskForm onSubmit={editTask} onSubmitName="Edit task" task={task} />
        </div>
      </div>
    </Modal>
  );
};

TaskModal.defaultProps = {
  task: { title: "", description: "" },
};

TaskModal.propTypes = {
  isOpen: bool.isRequired,
  onClose: func.isRequired,
  editTask: func.isRequired,
  task: shape({
    title: string,
    description: string,
  }),
};

export default TaskModal;
