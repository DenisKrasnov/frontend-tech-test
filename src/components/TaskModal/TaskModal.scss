.TaskModal {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 2000;
  display: flex;
  align-items: center;
  justify-content: center;
}

.backdrop {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(9, 15, 41, 0.7);
}

.modal {
  position: relative;
  width: 500px;
  max-height: calc(100vh - 40px);
  overflow-y: auto;
  outline: none;
  background-color: #fff;
  box-shadow: 0 4px 16px rgba(9, 15, 41, 0.4);
}

.header {
  height: 20px;
  padding: 20px;
  text-align: left;
}

.title {
  font-weight: 600;
  display: inline;
}

.closeButton {
  float: right;
  cursor: pointer;
}

.body {
  padding: 20px;
  text-align: left;
}

.footer {
  background: $main-bg-color;
  border: 1px solid $border-color;
  text-align: right;
  padding: 20px;
  user-select: none;
}
