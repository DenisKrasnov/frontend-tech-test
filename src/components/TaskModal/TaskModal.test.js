import React from "react";
import { shallow } from "enzyme";

import TaskModal from "./TaskModal";

const noop = () => {};

describe("TaskModal", () => {
  it("renders", () => {
    const component = shallow(
      <TaskModal editTask={noop} isOpen onClose={noop} />
    );
    expect(component).toMatchSnapshot();
  });
});
