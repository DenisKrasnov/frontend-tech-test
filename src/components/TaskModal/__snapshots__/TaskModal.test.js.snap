// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskModal renders 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropClassName="backdrop"
  className="TaskModal"
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <div
    className="modal"
  >
    <div
      className="header"
    >
      <h2
        className="title"
      >
        Edit task
      </h2>
      <span
        className="closeButton"
      >
        <Icon
          color="blue"
          glyph="x"
          onClick={[Function]}
        />
      </span>
    </div>
    <div
      className="body"
    >
      <TaskForm
        headerName=""
        loading={false}
        onSubmit={[Function]}
        onSubmitName="Edit task"
        task={
          Object {
            "description": "",
            "title": "",
          }
        }
        validate={[Function]}
      />
    </div>
  </div>
</Modal>
`;
